src/main/java/de/zalando/sprocwrapper/SProcCall.java

```diff
@@ -41,10 +41,35 @@ public long getSprocId() {
     }
 
     public static enum WriteTransaction {
-        USE_FROM_SERVICE,
-        NONE,
-        ONE_PHASE,
-        TWO_PHASE
+        USE_FROM_SERVICE {
+            @Override
+            public SProcService.WriteTransaction getServiceWriteTransaction(SProcService.WriteTransaction serviceWriteTransaction) {
+                if (serviceWriteTransaction == null) {
```

Comment:
It's my question too. I think SprocCall does not need WriteTransaction enum. and my first idea was removing it completely.
In my book, the main reason that we cannot find a common sense about solution is its unnecessity.
But i thought maybe we miss some backward compatibility. If it is not a case. Let me know. Then i will remove it instantly.